### PR TITLE
translation: Update chapter_stack_and_queue/summary.md

### PR DESCRIPTION
**Pull request by QiLOL**
If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [ ] I've tested the code and ensured the outputs are the same as the outputs of reference code.
- [ ] I've checked the code (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

**Comment by krahets** [comment]
Randomly assign ['@yuelinxin' '@SamJin98'] to review the pull request #1048

File: docs-en/chapter_stack_and_queue/summary.md [hunk 1]
```diff
@@ -2,29 +2,29 @@
...
-- Regarding space efficiency, the array implementation of a stack may lead to some level of space wastage. However, it's important to note that the memory space occupied by nodes in a linked list is generally larger than that for elements in an array.
-- A queue is a data structure that follows the First-In-First-Out (FIFO) principle, and it can also be implemented using either arrays or linked lists. The conclusions regarding time and space efficiency for queues are similar to those for stacks.
-- A double-ended queue is a more flexible type of queue that allows adding and removing elements from both ends.
+- Stack is a data structure that follows the Last-In-First-Out (LIFO) principle and can be implemented using arrays or linked lists.
+- In terms of time efficiency, the array implementation of a stack has a higher average efficiency. However, during expansion, the time complexity for a single push operation can degrade to $O(n)$. In contrast, the linked list implementation of a stack offers more stable efficiency.
```
Comment (yuelinxin) [inline docs-en/chapter_stack_and_queue/summary.md:6]: "the array implementation of stack has higher average efficiency" might be cleaner.

File: docs-en/chapter_stack_and_queue/summary.md [hunk 2]
```diff
@@ -2,29 +2,29 @@
...
+- Regarding space efficiency, the array implementation of a stack may lead to a certain degree of space wastage. However, it's important to note that the memory space occupied by nodes in a linked list is generally larger than that for elements in an array.
+- A queue is a data structure that follows the First-In-First-Out (FIFO) principle, and it can also be implemented using arrays or linked lists. The conclusions regarding time and space efficiency for queues are similar to those for stacks.
+- A double-ended queue(deque) is a more flexible type of queue that allows adding and removing elements at both ends.
```
Comment (yuelinxin) [inline docs-en/chapter_stack_and_queue/summary.md:9]: please add a space between "(" and the leading characters.

File: docs-en/chapter_stack_and_queue/summary.md [hunk 3]
```diff
@@ -2,29 +2,29 @@
...
 
 ### Q & A
 
 **Q**: Is the browser's forward and backward functionality implemented with a doubly linked list?
 
-The forward and backward functionality of a browser fundamentally represents the "stack" concept. When a user visits a new page, it is added to the top of the stack; when they click the back button, the page is popped from the top. A double-ended queue can conveniently implement some additional operations, as mentioned in the "Double-Ended Queue" section.
+A browser's forward and backward navigation is essentially a manifestation of the "stack" concept. When a user visits a new page, the page is added to the top of the stack; when they click the back button, the page is popped from the top of the stack. A double-ended queue(deque) can conveniently implement some additional operations, as mentioned in the "Double-Ended Queue" section.
```
Comment (yuelinxin) [inline docs-en/chapter_stack_and_queue/summary.md:15]: please add a space between "(" and the leading characters.

**Review by yuelinxin — COMMENTED** [review]
Please consider the following suggestions.

File: docs-en/chapter_stack_and_queue/summary.md [hunk 4]
```diff
@@ -2,29 +2,29 @@
...
 
 **Q**: After popping from a stack, is it necessary to free the memory of the popped node?
 
-If the popped node will still be used later, it's not necessary to free its memory. In languages like Java and Python that have automatic garbage collection, manual memory release isn't required; in C and C++, manual memory release is necessary if the node will no longer be used.
+If the popped node will still be used later, it's not necessary to free its memory. In languages like Java and Python that have automatic garbage collection, manual memory release is not necessary; in C and C++, manual memory release is required.
 
 **Q**: A double-ended queue seems like two stacks joined together. What are its uses?
 
-A double-ended queue is essentially a combination of a stack and a queue, or like two stacks joined together. It exhibits both stack and queue logic, therefore enabling the implementation of all applications of stacks and queues with added flexibility.
+A double-ended queue is a combination of a stack and a queue or two stacks joined together. It exhibits both stack and queue logic. Thus, it can implement all applications of stacks and queues, offering more flexibility.
```
Comment (SamJin98) [inline docs-en/chapter_stack_and_queue/summary.md:23]: You could connect two sentences to make them coherent:
"A double-ended queue is a combination of a stack and a queue or two stacks joined together. It exhibits both stack and queue logic." -> "A double-ended queue, which is a combination of a stack and a queue or two stacks joined together, exhibits both stack and queue logic."

**Review by SamJin98 — COMMENTED** [review]
Thanks for the PR, here's one improvement you could take.

**Comment by QiLOL** [comment]
@yuelinxin @SamJin98 Thanks for the suggestions, the doc is updated.

**Review by SamJin98 — APPROVED** [review]

**Review by yuelinxin — APPROVED** [review]
Looks good to me! Good work.

**Review by krahets — APPROVED** [review]

**Pull request merged**